app/frontend/actions.ts

```diff
@@ -553,10 +553,8 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
   const resetDelegation = () => {
     setState({
       shelleyDelegation: {
-        delegationFee: 0,
-        selectedPool: {
-          poolHash: '',
-        },
+        delegationFee: 0 as Lovelace,
+        selectedPool: null,
```

Comment:
Maybe we could reset the delegation to something more sensible. For example if we reset it to 
```
{
    poolId: '',
    ticker: '',
    ...
}
```
Not sure this wont cause some issues but at least we would not have to check whether this are defined or selectedPool is null etc. but we would instead always display these.

Reply:
I don't like this approach that much. Idea is that when you see object that is `null`, in this case `selectedPool` you know that it doesn't exists - you don't have anything selected. Having a non-null field called `selectedPool` is giving impression, that you in have a pool selected.

Also, maybe not that related to this, but I also quite dislike that we have `validationError` inside of this object.

Reply:
I agree about the null and selected pool. In case we want it to rest it to empty string we should at least rename it to `selectedPoolInfo` or something similar. Anyway lets keep the null here. 
regarding the error inside the pools, previously there used to be multiple pools you could delegate to so this is what is left of it. Feel free to refactor this but probably in another PR since we don't want to prolong this even more.